src/Auth.ts

```diff
@@ -82,34 +85,45 @@ export class Auth {
   /**
    * Returns the user data of the JWT decoded payload
    */
-  async getPayload() {
-    if (!this.isLoggedIn()) {
-      await this.login()
-    }
-
-    const payload = jwt.decode(this.accessToken!)
-    return payload
+  async getAccessTokenData() {
+    return jwt.decode(await this.getAccessToken()) as AccessToken
   }
 
-  async getToken() {
+  async getAccessToken(): Promise<string> {
     if (this.accessToken) {
       try {
-        const publicKey = await this.getServerPublicKey()
-        jwt.verify(this.accessToken, publicKey)
-        return this.accessToken
+        const tokenData = jwt.decode(this.accessToken) as AccessToken
+        const currentPubKey = this.ephemeralKey!.key.publicKeyAsHexString()
```

Comment:
let's take this outside of the if and use it as the pubKey below

Reply:
If the user is not log in, and we never call `getPublicKey()`  `this.ephemeralKey` could be null, so doing `this.ephemeralKey!.key.publicKeyAsHexString()` outside the if would throw an error.

Having said that, i think we should generate the ephKey in the constructor, this would leave the Auth() object in a valid initial state.  Then we can do what you suggest, agree???

Reply:
I don't mind moving it to the constructor, but maybe there are reasons for doing it here, like if you want to have an Auth in a test but you don't require a pubkey.

I think I got confused, I wanted to check the key for this case and the case below, but since we ask a for a new access token with the pubkey obviously is going to match. Sorry about that, I think what you did is ok

Reply:
```const currentPubKey = this.ephemeralKey!.key.publicKeyAsHexString()```

This is using an ephemeralKey that may be expired. If the public key matches the one used to get the access token, then the `if` below will pass, and we will return an access token from an expired ephemeral key.

Why don't we use `this.getEphemeralKey()` instead? If the key is expired, that will return a new ephemeralKey, and the `if` below would not pass, and a new access token will be generated with this valid ephemeral key.

Reply:
we can do something like moving the `this.getEphemeralKey()` at the top of the `getAccessToken()` method. In that case if the key expired, the current `accessToken` will get invalidated, and we will generate a new one with the new ephKey that we just generated